internal/cmd/launcher.go

```diff
@@ -90,29 +93,30 @@ func (l *Launcher) Launch() {
 			" The current k6 binary doesn't satisfy all dependencies, it's required to" +
 			" provision a custom binary.")
 
-	customBinary, err := l.provision(l.gs, deps)
+	customBinary, err := l.provider.provision(deps)
 	if err != nil {
 		l.gs.Logger.
 			WithError(err).
 			Error("Failed to provision a k6 binary with required dependencies." +
 				" Please, make sure to report this issue by opening a bug report.")
-		l.gs.OSExit(1)
-		return
+		return err
 	}
 
-	customBinary.run(l.gs)
+	l.commandExecutor = customBinary
+
+	// override command's RunE method to be processed by the command executor
+	cmd.RunE = l.runE
+
+	return nil
 }
 
-// customBinary runs the requested commands
-// on a different binary on a subprocess passing the original arguments
-type customBinary struct {
-	// path represents the local file path
-	// on the file system of the binary
-	path string
```

Comment:
Can we restore this struct definition to keep the same position as before? I don't see the reason to separate the struct and `run` method definitions.